### PR TITLE
Add flutter run support for linux and windows

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Adds buildable application packages for Windows and Linux, and allows running them.

## Related Issues

https://github.com/flutter/flutter/issues/30707

## Tests

I added the following tests:
- Updated defaults for Linux and Windows test

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Haven't tested the windows logic end to end yet, getting my home windows machine set up for it currently.

**Review by stuartmorgan-g — COMMENTED** [review]
Working through issues with Windows. So far:
- project.dart accidentally used name_output.sh instead of name_output.bat, which we should fold a fix for into this patch
- "Cannot find executable for Get-CimInstance Win32_Process". Is processManager.runSync running things in PowerShell?

**Comment by jonahwilliams** [comment]
Updated the process script to try and run through power shell, fixed the typo in the windows project. Time to write tests.

File: packages/flutter_tools/lib/src/linux/application_package.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  final LinuxProject project;
+
+  @override
+  String executable(BuildMode buildMode) {
+    final ProcessResult result = processManager.runSync(<String>[
+      project.nameScript.path,
+      buildMode == BuildMode.debug ? 'debug' : 'release',
+    ]);
+    if (result.exitCode != 0) {
+      throwToolExit('Failed to find Windows project name');
```
Comment (stuartmorgan) [inline packages/flutter_tools/lib/src/linux/application_package.dart]: s/Windows/Linux/

File: packages/flutter_tools/lib/src/windows/windows_device.dart [hunk 2]
```diff
@@ -99,3 +135,45 @@ class WindowsDevices extends PollingDeviceDiscovery {
...
 }
+
+final RegExp _whitespace = RegExp(r'\w+');
+
+/// Returns the running process matching `process` name.
+///
+/// This list contains the process name and id.
+@visibleForTesting
+List<String> runningProcess(String processName) {
+  // Does this only work from powershell?
```
Comment (stuartmorgan) [inline packages/flutter_tools/lib/src/windows/windows_device.dart]: Should this be a TODO to find a non-shell way to do this?

**Review by stuartmorgan-g — APPROVED** [review]

**Pull request merged**